Mathlib/RingTheory/IntegralClosure/Algebra/Basic.lean

```diff
@@ -243,3 +243,22 @@ instance Algebra.IsIntegral.tensorProduct [CommRing B]
   isIntegral p := p.induction_on isIntegral_zero (fun _ s ↦ .tmul _ <| int.1 s) (fun _ _ ↦ .add)
 
 end TensorProduct
+
+section MulSemiringAction
+
+variable (R K : Type*) [CommRing R] [CommRing K] [Algebra R K]
+    (G : Type*) [Group G] [MulSemiringAction G K] [SMulCommClass G R K]
+
+instance : MulSemiringAction G (integralClosure R K) where
+  smul := fun g x ↦ ⟨g • (x : K), x.2.map (MulSemiringAction.toAlgHom R K g)⟩
```

Comment:
Can you provide a `coe_smul` simp lemma and a `SMulCommClass G R (integralClosure R K)` instance? Thanks.